students/TinaB/lesson04/mailroom_part2.py

```diff
@@ -0,0 +1,231 @@
+        thank_you_letter = print_thank_you(donor_name[0], donor_amount)
+        print(thank_you_letter)
+
+
+def print_donors_names():
+    """ prints list of donors"""
+    print()
+    for d in donors_list_dictionary:
+        print("{firstname} {lastname}".format(**d))
+    print()
+
+
+def print_thank_you(donor_fname, amount):
+    """ prints thank you message"""
+    d = {'name': donor_fname, 'donation': amount}
+    thank_you = '\nDear {},\n '.format(donor_fname)
+    thank_you += '\tThank you for your generous donation of ${donation:,.2f}\n'.format(
+        **d)
+    thank_you += 'Sincerely, \nThe ChickTech Donations Department\n'
+    return thank_you
+
+
+def print_thank_you_total(donor):
+    """ prints thank you message"""
+    #d = {'name': donor_fname, 'donation': amount}
+    donor_output = {"firstname": donor["firstname"],"lastname": donor["lastname"] }
+    all_donations = donor["donations"]
+    donor_output['last_donation'] = all_donations[len(all_donations) - 1]
+    total = 0
+    for gift in all_donations:
+        total += gift
+    donor_output['total_donations'] = total
+    thank_you = '''\nDear {firstname} {lastname}\n
+        \t Thank you for your most recent generous donation of ${last_donation:,.2f}\n
+        \t You're support of {total_donations} over the years has helped us fund many great programs!
+        \t Again, thank you! We love your support.
+            Sincerely, 
+            \n The ChickTech Donations Department\n
+    '''.format(**donor_output)
+    return thank_you
+
+
+def print_report():
+    """Print report to match example from assignment for donor list """
+    print()
+    title = ['Donor Name', '|  Total Given ', '|   Num Gifts',
+             '  | Average Gift']
+    print('{:<20}{:>14}{:^14}{:>14}'.format(title[0], title[1],
+                                            title[2], title[3]))
+    print('-'*65)
+    print()
+    # Creating list to hold donors info for printing
+    donor_list = list()
+    for donor in donors_list_dictionary:
+        donor_fullname = "{firstname} {lastname}".format(**donor)
+        # donor object will hold fullname, donation total, donation times, average donation
+        donor_info = [donor_fullname, 0, 0, 0]
+        for donor_amount in donor["donations"]:
+            donor_info[1] += donor_amount
+            donor_info[2] += 1
+        donor_info[3] = donor_info[1] // donor_info[2]
+        donor_list.append(donor_info)
+
+        print('{:<22}{}{:>12.2f}{:>10}{:>8}{:>12.2f}'.format(donor_fullname, '$',
+                                                             donor_info[1], donor_info[2], '$', donor_info[3]))
+    print()
+
+
+def send_letters_everyone():
+    """Creates a letter for everyone in the database, and writes them to file."""
+    letters_count = 0
+    new_folder = str(datetime.datetime.now())
+    try:
+        os.mkdir(new_folder)
+    except OSError as oserr:
```

Comment:
You don't need to cast this as `oserr` unless you're going to use the `osser` variable in your `except` block